### ISSUE TITLE
Very strange error in conact function with dollar sign

### ISSUE THREAD

**Issue by moonjoungyoung** [sequelize/sequelize#13772]
## Issue Creation Checklist

[X] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### SSCCE

**Here is the link to the SSCCE for this issue:** LINK-HERE 

```js
// model
  model.beforeFind((options) => {
    if (options.where && options.where.password && typeof (options.where.password) === 'string') {
      console.log('options.where.password', options.where.password);
      options.where.password = sequelize.fn('CONCAT', options.where.password, sequelize.col('id'));
    }
  });

// controller
  await Model.findAll({
    where: {
      id: 123,
      password: '123$123',
    },
    logging: console.log,
  });

// console
options.where.password 123$123

Executing (default):
SELECT `id`, `password`, `deletedAt`
FROM `_person` AS `_person`
WHERE (`_person`.`deletedAt` IS NULL AND (`_person`.`id` = '123' AND `_person`.`password` = CONCAT('123$$123', `id`)));
```

### What do you expect to happen?

I expect SQL seems ``CONCAT('123$123', `id`)``

### What is actually happening?

SQL is ``CONCAT('123$$123', `id`)``;
ONE DOLLAR SIGN HAS ADDED!!

### Additional context

Add any other context and details here.

### Environment

- Sequelize version: 6.9.0 
- Node.js version: 12.22.3 

### How does this problem relate to dialects?

- [X] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [X] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by WikiRik** [comment]
Upon initial analysis, this is due to the string with the dollar sign being one of the fn arguments. Will have to investigate further, but I can reproduce this error.

**Comment by WikiRik** [comment]
This is due to https://github.com/sequelize/sequelize/pull/11606. Not sure if this is something we can fix properly, but there is a workaround in the form of [sequelize.literal()](https://sequelize.org/v6/class/lib/sequelize.js~Sequelize.html#static-method-literal)

So you can use
```js
options.where.password = sequelize.fn('CONCAT', sequelize.literal(options.where.password), sequelize.col('id'));
```

**Comment by WikiRik** [comment]
Closed in favor of #12523
